### PR TITLE
SAMZA-1193. re-enable the test TestZkBarrierForVersionUpgrade.testZkBarrierForVersionUpgrade

### PR DESCRIPTION
**Pull request by sborya**
The code changed significantly. I've rerun the test multiple times both with gradle and intelij. It passed every time. I suggest we enable it back.

### CODE REVIEW

**Comment by navina** [comment]
@sborya what was the change done here, other than changing the barrier name in the tests?

File: samza-core/src/test/java/org/apache/samza/zk/TestZkBarrierForVersionUpgrade.java [hunk 1]
```diff
@@ -74,8 +73,7 @@ public static void teardown() {
     zkServer.teardown();
```
Comment (fredji97) [inline samza-core/src/test/java/org/apache/samza/zk/TestZkBarrierForVersionUpgrade.java]: zkServer.setup is called in Before, but zkServer.teardown is called in AfterClass. You may want to call zkServer.teardown in After for each test, like  zkServer.setup in Before, or make zkServer.setup called in BeforeClass.

**Comment by sborya** [comment]
Moved init stuff into Before and After methods.

File: samza-core/src/test/java/org/apache/samza/zk/TestZkBarrierForVersionUpgrade.java (hunk 1 above)
Comment (sborya) [inline samza-core/src/test/java/org/apache/samza/zk/TestZkBarrierForVersionUpgrade.java]: Good point. Fixed.

Comment (fredji97) [inline samza-core/src/test/java/org/apache/samza/zk/TestZkBarrierForVersionUpgrade.java]: thanks for addressing it!

**Pull request closed without merging**